### DISCUSSION TITLE
Paring api Metadata add custom field for extra data dict.

### DISCUSSION THREAD

**Discussion by hamberluo** [WalletConnect/walletconnect-monorepo#5163] [Ideas]
We need process extra data to do our own logic.

```json
{
  "name": string,
  "description": string,
  "url": string,
  "icons": [string],
  "verifyUrl": string, // Optional
  "redirect": { // Optional
    "native": string, // Optional
    "universal": string, // Optional
  },
  "extra": {}
}
```

**Comment by glitch-txs** [comment]
hello could you expand on why do you need this field?

**Reply by stepanLav** [reply]
Hello!

I'm writing from Novasama Technologies to express our support for the feature request to exchange additional metadata (like wallet names) during the pairing process, as outlined in this discussion.

This capability would be incredibly valuable for our ecosystem as well. Specifically, it would enable a smoother user experience when integrating our products, [Nova Spektr](https://novaspektr.io/) and [Nova Wallet](https://novawallet.io/), by allowing for a clearer exchange of information during the connection handshake.

Also provide an example how it can be implemented:
https://github.com/WalletConnect/walletconnect-monorepo/pull/6922
